aws-opentelemetry-distro/tests/amazon/opentelemetry/distro/test_instrumentation_patch.py

```diff
@@ -115,6 +145,149 @@ def _test_patched_botocore_instrumentation(self):
         self.assertTrue("aws.sqs.queue_name" in sqs_attributes)
         self.assertEqual(sqs_attributes["aws.sqs.queue_name"], _QUEUE_NAME)
 
+        # Bedrock
+        bedrock_sucess_attributes: Dict[str, str] = _do_bedrock_on_success()
+        self.assertTrue("aws.bedrock.guardrail_id" in bedrock_sucess_attributes)
+        self.assertEqual(bedrock_sucess_attributes["aws.bedrock.guardrail_id"], _BEDROCK_GUARDRAIL_ID)
+
+        # Bedrock Agent Operation
+        self.assertTrue("bedrock-agent" in _KNOWN_EXTENSIONS)
+        bedrock_agent_op_attributes: Dict[str, str] = _do_extract_bedrock_agent_op_attributes()
+        self.assertTrue("aws.bedrock.agent_id" in bedrock_agent_op_attributes)
+        self.assertEqual(bedrock_agent_op_attributes["aws.bedrock.agent_id"], _BEDROCK_AGENT_ID)
+        bedrock_agent_op_sucess_attributes: Dict[str, str] = _do_bedrock_agent_op_on_success()
+        self.assertTrue("aws.bedrock.agent_id" in bedrock_agent_op_sucess_attributes)
+        self.assertEqual(bedrock_agent_op_sucess_attributes["aws.bedrock.agent_id"], _BEDROCK_AGENT_ID)
+
+        # Bedrock DataSource Operation
+        self.assertTrue("bedrock-agent" in _KNOWN_EXTENSIONS)
+        bedrock_datasource_op_attributes: Dict[str, str] = _do_extract_bedrock_datasource_op_attributes()
+        self.assertTrue("aws.bedrock.datasource_id" in bedrock_datasource_op_attributes)
+        self.assertEqual(bedrock_datasource_op_attributes["aws.bedrock.datasource_id"], _BEDROCK_DATASOURCE_ID)
+        bedrock_datasource_op_sucess_attributes: Dict[str, str] = _do_bedrock_datasource_op_on_success()
+        self.assertTrue("aws.bedrock.datasource_id" in bedrock_datasource_op_sucess_attributes)
+        self.assertEqual(bedrock_datasource_op_sucess_attributes["aws.bedrock.datasource_id"], _BEDROCK_DATASOURCE_ID)
+
+        # Bedrock KnowledgeBase Operation
+        self.assertTrue("bedrock-agent" in _KNOWN_EXTENSIONS)
+        bedrock_knowledgebase_op_attributes: Dict[str, str] = _do_extract_bedrock_knowledgebase_op_attributes()
+        self.assertTrue("aws.bedrock.knowledgebase_id" in bedrock_knowledgebase_op_attributes)
+        self.assertEqual(bedrock_knowledgebase_op_attributes["aws.bedrock.knowledgebase_id"], _BEDROCK_KNOWLEDGEBASE_ID)
+
+        # Bedrock Agent Runtime
+        self.assertTrue("bedrock-agent-runtime" in _KNOWN_EXTENSIONS)
+        bedrock_agent_runtime_attributes: Dict[str, str] = _do_extract_bedrock_agent_runtime_attributes()
+        self.assertTrue("aws.bedrock.agent_id" in bedrock_agent_runtime_attributes)
+        self.assertEqual(bedrock_agent_runtime_attributes["aws.bedrock.agent_id"], _BEDROCK_AGENT_ID)
+        self.assertTrue("aws.bedrock.knowledgebase_id" in bedrock_agent_runtime_attributes)
+        self.assertEqual(bedrock_agent_runtime_attributes["aws.bedrock.knowledgebase_id"], _BEDROCK_KNOWLEDGEBASE_ID)
+
+        # BedrockRuntime
```

Comment:
There is a lot of code duplication here. Break this out into a helper function as I recommend above for bedrock, bedrock-agent, and bedrock-agent-runtime. Then do the following:
1. Define a set of bodies (e.g. titan_populated, titan_empty, claude_populated, claude_empty) that you can use with a simple set of helpers
2.
```
expected_extract_attribute_keys = [gen_ai.system, gen_ai.request.model, gen_ai.request.top_p ...] 
extract_populated = _do_extract_bedrock_runtime("titan", titan_populated)
extract_empty = _do_extract_bedrock_runtime("titan", titan_empty)
on_success_populated = _on_success_bedrock_runtime("titan", true)
on_success_empty = _on_success_bedrock_runtime("titan", false)
validate_attributes(extract_populated, expected_extract_attribute_keys, extract_empty, on_success_populated, on_success_empty)

expected_extract_attribute_keys = [gen_ai.system, gen_ai.request.model] 
extract_populated = _do_extract_bedrock_runtime("claude", claude_populated)
extract_empty = _do_extract_bedrock_runtime("claude", claude_empty)
on_success_populated = _on_success_bedrock_runtime("claude", true)
on_success_empty = _on_success_bedrock_runtime("claude", false)
validate_attributes(extract_populated, expected_extract_attribute_keys, extract_empty, on_success_populated, on_success_empty)
...
```

Reply:
I reformat to simply check `gen_ai.system `and` gen_ai.request.model` since we will remove runtime 6 remaining attributes. Will add back for next PR.